sql/core/src/main/scala/org/apache/spark/sql/jdbc/DB2Dialect.scala

```diff
@@ -207,7 +207,7 @@ private case class DB2Dialect() extends JdbcDialect with SQLConfHelper with NoLe
       val offsetClause = dialect.getOffsetClause(offset)
 
       options.prepareQuery +
-        s"SELECT $hintClause$columnList FROM ${options.tableOrQuery} $tableSampleClause" +
```

Comment:
I have a little question. For `tableSample`, currently only `DatabricksDialect` and `PostgresDialect` are supported, but I see that the `tableSampleClause` is also added in `MySQLSQLQueryBuilder` and `MsSqlServeSQLQueryBuilder`.

Reply:
In addition, if `hintClause` is not called in `MySQLSQLQueryBuilder` and `DB2SQLBuilder`, if users set `hint` option when using `MsSqlServerDialect` and `DB2Dialect`, will no error be thrown?

Reply:
Hmm, we need to remove them.

Reply:
@wayneguow Do you interested in fix it?

Reply:
BTW, that could be an orthogonal topic, @beliefer and @wayneguow . :)